### PR TITLE
Upgrade activerecord to 5.1.1

### PR DESCRIPTION
**Pull request by Davidslv**
ActiveRecord::ConnectionAdapters::ConnectionManagement has been
deprecated.

ActiveRecord::QueryCache no longer receives app on initialization since
https://github.com/rails/rails/commit/d3c9d808e3e242155a44fd2a89ef272cfade8fe8#diff-31e798bb03e4140196a88e242f488880L26

Dependencies:
- [x] https://trello.com/c/XQgx7sdX/86-upgrade-transition-to-a-supported-version-of-rails

Trello:
https://trello.com/c/fQ4AUcFA/88-upgrade-bouncer-to-a-supported-version-of-rails

### CODE REVIEW

**Comment by Davidslv** [comment]
@carvil I've added the functionality as explained in the conversation of that link you provided.

File: lib/connection_management.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# This functionality has been removed from ActiveRecord
+# see: https://github.com/rails/rails/issues/26947#issuecomment-265372902
+class ConnectionManagement
```
Comment (carvil) [inline lib/connection_management.rb]: What do you think of leaving the namespace `ActiveRecord::Rack::` in here?

**Comment by carvil** [comment]
This looks good @Davidslv , thanks.

Let's wait on the Transition upgrade before we merge it.

**Comment by Davidslv** [comment]
This has been deployed in integration and I haven't seen any problems. Just waiting for Transition now.

File: Gemfile.lock [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
-      activesupport (= 4.2.7.1)
-      builder (~> 3.1)
-    activerecord (4.2.7.1)
-      activemodel (= 4.2.7.1)
-      activesupport (= 4.2.7.1)
-      arel (~> 6.0)
-    activesupport (4.2.7.1)
+    activemodel (5.1.0)
+      activesupport (= 5.1.0)
+    activerecord (5.1.0)
```
Comment (carvil) [inline Gemfile.lock]: @Davidslv could you please upgrade to `5.1.1`? This should be the same version as `transition`: https://github.com/alphagov/transition/blob/master/Gemfile.lock#L43

**Pull request merged**